File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
import java.io.*;

class Solution {
    public int[] solution(int[] answers) {
        
        
        int[] person1 = {1,2,3,4,5};
        int[] person2 = {2,1,2,3,2,4,2,5};
        int[] person3 = {3,3,1,1,2,2,4,4,5,5};
        
        int temp1 = 0;
        int temp2 = 0;
        int temp3 = 0;
        
        for(int i=0; i<answers.length; i++){
            if(answers[i] == person1[i%person1.length]){
                temp1++;
            }
            if(answers[i] == person2[i%person2.length]){
                temp2++;
            }
            if(answers[i] == person3[i%person3.length]){
                temp3++;
            }
        }
        
        ArrayList<Integer> tempArray = new ArrayList<>();
        
        int temp = Math.max(temp1,Math.max(temp2,temp3));
        
        if(temp1 == temp){
            tempArray.add(1);
        }
        if(temp2 == temp){
            tempArray.add(2);
        }
        if(temp3 == temp){
            tempArray.add(3);
        }
        
        int[] answer = new int[tempArray.size()];
        
        for(int i=0; i<tempArray.size(); i++){
            answer[i] = tempArray.get(i);
        }
        
        Arrays.sort(answer);
        
        return answer;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    
    let person1 = [1,2,3,4,5];
    let person2 = [2,1,2,3,2,4,2,5];
    let person3 = [3,3,1,1,2,2,4,4,5,5];
    let score = [0,0,0];
    
    for(let i=0; i<answers.length;i++){
        if(answers[i] == person1[i % person1.length]){
            score[0] += 1;
        }
        if(answers[i] == person2[i % person2.length]){
            score[1] += 1;
        }
        if(answers[i] == person3[i % person3.length]){
            score[2] += 1;
        }
    }
    
    let maxScore = Math.max(...score);
    
    let answer = [];
    
    for(let i=1; i<=score.length;i++){
        if(score[i-1] == maxScore){
            answer.push(i);
        }
    }
    return answer;
    
}